### ISSUE TITLE
Investigate Google Vision API for OCR

### ISSUE THREAD

**Issue by jashkenas** [documentcloud/documentcloud#406]
You guys might have already looked at this — but it seems like the Google Vision API provides really stellar OCR output. It might be worth seeing if their terms and rate limits would allow you to use their OCR as an alternative to Docsplit's Tesseract OCR.

Obviously, for sensitive documents you don't want to send to google, perhaps the default would remain the same. But having an option to "give me really good OCR" at upload would be lovely.

https://cloud.google.com/vision/

**Comment by knowtheory** [comment]
yeah, i'd love to have pluggable backends for OCRing (including producing & stashing hOCR data) and data processing in general.

For example I've got prototype [implementation of a tabula integration](https://github.com/documentcloud/documentcloud/blob/master/app/actions/extract_tables.rb)

One of the things i haven't had time to throw brain power at is what a reasonable general ReSTful layout for the resulting output for jobs like this should be.  A tabula table scraper should probably dump into some sort of namespaced layout? 

And where in the ReST hierarchy the namespace should fall:

`/:id/pages/:namespace/:title-p:page_number.csv`
`/:id/:namespace/pages/:title-p:page_number.csv`

**Comment by jashkenas** [comment]
Tabula would be lovely, but page OCR would be far easier ;)

Page-image in -> Plain text out. Stuff it in the DB and you're done.
